src/compiler/crystal/macros/methods.cr

```diff
@@ -1603,7 +1603,7 @@ module Crystal
       when "struct?"
         interpret_check_args { BoolLiteral.new(type.class? && type.struct?) }
       when "nilable?"
-        interpret_check_args { BoolLiteral.new(type.nilable?) }
+        interpret_check_args { BoolLiteral.new(type.program.nil_type.implements?(type)) }
```

Comment:
I'm wondering if we shouldn't keep the `#nilable?` method, with the new implementation. There are many more methods like it, some with even more trivial implementation. So it feels odd to remove this one.